File: app/src/main/java/com/unbounds/trakt/widget/TintImageView.java
package com.unbounds.trakt.widget;

import android.content.Context;
import android.content.res.ColorStateList;
import androidx.appcompat.widget.AppCompatImageView;
import android.util.AttributeSet;

/**
 * Created by maclir on 2015-11-21.
 */
public class TintImageView extends AppCompatImageView {

	private ColorStateList tint;

	public TintImageView(final Context context) {
		super(context);
	}

	public TintImageView(final Context context, final AttributeSet attrs) {
		super(context, attrs);
	}

	public TintImageView(final Context context, final AttributeSet attrs, final int defStyle) {
		super(context, attrs, defStyle);
	}

	@Override
	protected void drawableStateChanged() {
		super.drawableStateChanged();
		if (tint != null && tint.isStateful())
			updateTintColor();
	}

	private void updateTintColor() {
		final int color = tint.getColorForState(getDrawableState(), 0);
		setColorFilter(color);
	}

}

File: app/src/main/java/com/unbounds/trakt/view/widget/TintImageView.kt
package com.unbounds.trakt.view.widget

import android.content.Context
import android.content.res.ColorStateList
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView

/**
 * Created by maclir on 2015-11-21.
 */
class TintImageView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = 0) : AppCompatImageView(context, attrs, defStyle) {
    private val tint: ColorStateList? = null

    override fun drawableStateChanged() {
        super.drawableStateChanged()
        if (tint != null && tint.isStateful) updateTintColor()
    }

    private fun updateTintColor() {
        val color = tint!!.getColorForState(drawableState, 0)
        setColorFilter(color)
    }
}
